internal/mode/static/state/graph/graph.go

```diff
@@ -46,6 +49,8 @@ type Graph struct {
 	ReferencedSecrets map[types.NamespacedName]*Secret
 	// ReferencedNamespaces includes Namespaces with labels that match the Gateway Listener's label selector.
 	ReferencedNamespaces map[types.NamespacedName]*v1.Namespace
+	// ReferencedServicesNames includes the names of all the Services that are referenced by at least one HTTPRoute.
+	ReferencedServicesNames map[types.NamespacedName]struct{}
```

Comment:
ReferencedServicesNames will store a `NamespacedName` of all the `Services` that are referenced by any HTTPRoutes and does not guarantee that any of the `NamespacedName`'s are actual `Service`'s in the ClusterState. This will play into the logic in `IsReferenced` below. 

Also I saw no need to actually store the `Service` associated with the `NamespacedName`. Does that seem fine?

Reply:
> Also I saw no need to actually store the Service associated with the NamespacedName. Does that seem fine?

Yep 👍